prov/gni/test/rdm_sr.c

```diff
@@ -388,23 +389,22 @@ void do_sendmsg(int len)
 	sz = fi_recv(ep[1], target, len, rem_mr, gni_addr[0], source);
 	cr_assert_eq(sz, 0);
 
-	while ((ret = fi_cq_read(msg_cq[0], &cqe, 1)) == -FI_EAGAIN) {
-		pthread_yield();
-	}
-
-	cr_assert_eq(ret, 1);
-	cr_assert_eq((uint64_t)cqe.op_context, (uint64_t)target);
-
-	dbg_printf("got send context event!\n");
-
-	while ((ret = fi_cq_read(msg_cq[1], &cqe, 1)) == -FI_EAGAIN) {
-		pthread_yield();
-	}
+	/* need to progress both CQs simultaneously for rendezvous */
+	do {
+		ret = fi_cq_read(msg_cq[0], &s_cqe, 1);
+		if (ret == 1) {
+			source_done = 1;
+		}
+		ret = fi_cq_read(msg_cq[1], &d_cqe, 1);
+		if (ret == 1) {
+			dest_done = 1;
+		}
+	} while (!(source_done && dest_done));
 
-	cr_assert_eq(ret, 1);
-	cr_assert_eq((uint64_t)cqe.op_context, (uint64_t)source);
+	cr_assert_eq((uint64_t)s_cqe.op_context, (uint64_t)target);
+	cr_assert_eq((uint64_t)d_cqe.op_context, (uint64_t)source);
 
-	dbg_printf("got recv context event!\n");
+	dbg_printf("got context events!\n");
```

Comment:
All these prints should be disabled.  I used them for debugging when I first created the tests and again for this update.  I don't want to reinsert the messages if I make changes again.